src/engine/guidance/collapse_scenario_detection.cpp

```diff
@@ -147,6 +147,9 @@ bool isUTurn(const RouteStepIterator step_prior_to_intersection,
             step_prior_to_intersection, step_entering_intersection, step_leaving_intersection))
         return false;
 
+    if (hasTurnType(*step_entering_intersection), TurnType::OnRamp)
```

Comment:
Of course we want `OffRamp` as well. Thank you for pointing it out. It might be that other types are also of interest, although none come to mind right now. Added `OffRamp`.